mirage/config.js

```diff
@@ -621,6 +621,49 @@ export default function () {
     return this.serialize(builds, 'build');
   });
 
+  this.post('/repo/:id/request/config', (schema, { params }) => {
+    const requestConfig = schema.requestConfigs.create({
+      raw_configs: [{
+        source: 'test/test_repo:.travis.yml@master',
+        config: 'script: echo "Hello World"',
+        mode: 'deep_merge_append'
+      }],
+      request_config: {
+        language: 'node_js',
+        os: ['linux'],
+      },
+      job_configs: [{
+        os: 'linux',
+        language: 'node_js',
+      }],
+      messages: [
+        {
+          type: 'config',
+          level: 'info',
+          key: 'root',
+          code: 'default',
+          args: {
+            key: 'os',
+            default: 'linux'
+          }
+        }
+      ]
+    });
```

Comment:
I'd like to suggest creating a [mirage factory](https://www.ember-cli-mirage.com/docs/data-layer/factories) for `request-config` and make it build a record with default values. This would make it easier to adjust the records in the future. Also maybe some field can use [traits](https://www.ember-cli-mirage.com/docs/data-layer/factories#traits) to shape data.

In this case you'd just need to call `const requestConfig = server.create('request-config')` to create a record.

Reply:
makes sense.

i'll give this a try, but i'm not sure if i'm able to implement that. this was contributed by Emmanuel, so maybe i'll have to rely on him for this.

Reply:
This is totally optional, just a suggestion ;) If you don't have time for it, feel free to keep it as is, it works perfectly.